File: samples/nodejs/src/orderService.js
function calculateTotal(price, quantity, discountPercent) {
    if (quantity <= 0) throw new Error("Quantity must be positive");
    if (discountPercent < 0 || discountPercent > 100) throw new Error("Discount must be between 0 and 100");
    const subtotal = price * quantity;
    const discount = subtotal * (discountPercent / 100);
    return subtotal - discount;
}

function classifyOrder(total) {
    if (total >= 1000) return "LARGE";
    if (total >= 100)  return "MEDIUM";
    return "SMALL";
}

function isEligibleForFreeShipping(total, isPremiumMember) {
    return isPremiumMember || total >= 50;
}

module.exports = { calculateTotal, classifyOrder, isEligibleForFreeShipping };

File: samples/python/src/order_service.py
def calculate_total(price, quantity, discount_percent):
    if quantity <= 0:
        raise ValueError("Quantity must be positive")
    if not (0 <= discount_percent <= 100):
        raise ValueError("Discount must be between 0 and 100")
    subtotal = price * quantity
    discount = subtotal * (discount_percent / 100)
    return subtotal - discount


def classify_order(total):
    if total >= 1000:
        return "LARGE"
    if total >= 100:
        return "MEDIUM"
    return "SMALL"


def is_eligible_for_free_shipping(total, is_premium_member):
    return is_premium_member or total >= 50
